resources/js/Pages/Register.vue

```diff
@@ -0,0 +1,198 @@
+<script lang='ts'>
+  import Layout from '../Shared/Layout.vue'
+  import Vue from 'vue'
+
+  interface RegisterData {
+    firstname: string,
+    lastname: string,
+    username: string,
+    email: string,
+    organization: string,
+    password: string
+  }
+
+  export default Vue.extend({
+    name: 'Register',
+    components: {
+      Layout
+    },
+    data(): RegisterData {
+      return {
+        firstname: '',
+        lastname: '',
+        username: '',
+        email: '',
+        organization: '',
+        password: ''
+      }
+    },
+    computed: {
+      elements() {
+        return [
+          {
+            type: 'text',
+            name: 'firstname',
+            required: true,
+            placeholder: 'First Name',
+            value: this.firstname
+          },
+          {
+            type: 'text',
+            name: 'lastname',
+            required: true,
+            placeholder: 'Last Name',
+            value: this.lastname
+          },
+          {
+            type: 'text',
+            name: 'username',
+            required: true,
+            placeholder: 'Username',
+            value: this.username
+          },
+          {
+            type: 'text',
+            name: 'email',
+            required: true,
+            placeholder: 'Email',
+            value: this.email
+          },
+          {
+            type: 'text',
+            name: 'organization',
+            placeholder: 'Organization',
+            value: this.organization
+          },
+          {
+            type: 'password',
+            name: 'password',
+            required: true,
+            placeholder: 'Password',
+            value: this.password
+          }
+        ]
+      }
+    },
+    methods: {
+      getValidatedPayload(): string {
+        if (this.hasEmptyRequiredFields()) {
+          throw new Error('Please fill out all required fields.');
+        }
+
+        if (this.hasFieldsUnderFiveChars()) {
+          throw new Error('Please use field values greater than 5 characters.');
+        }
+
+        if (this.hasInvalidEmailFormat()) {
+          throw new Error('Please fill all required fields with at least 5 characters and valid email.');
```

Comment:
will reduce this to only mention email validity

Reply:
this is fine.